sdk/confidentialledger/Azure.Security.ConfidentialLedger/src/Generated/ConfidentialLedgerClient.cs

```diff
@@ -2661,5 +2770,85 @@ internal HttpMessage CreateGetUserDefinedFunctionsNextPageRequest(string nextLin
         private static ResponseClassifier ResponseClassifier201 => _responseClassifier201 ??= new StatusCodeClassifier(stackalloc ushort[] { 201 });
         private static ResponseClassifier _responseClassifier200201;
         private static ResponseClassifier ResponseClassifier200201 => _responseClassifier200201 ??= new StatusCodeClassifier(stackalloc ushort[] { 200, 201 });
+
+        // Format a GetLedgerEntry-shaped response from a GetCurrentLedgerEntry response body.
+        // Expected current entry body: { "collectionId":"...", "contents":"...", "transactionId":"..." }
+        // Desired ledger entry body: { "entry": { same fields }, "state": "Ready" }
+        private Response FormatLedgerEntry(Response currentResponse)
+        {
+            try
+            {
+                if (currentResponse?.ContentStream == null)
+                {
+                    return currentResponse; // nothing to do
+                }
+                currentResponse.ContentStream.Position = 0;
+                using (var doc = System.Text.Json.JsonDocument.Parse(currentResponse.ContentStream))
+                {
+                    var root = doc.RootElement;
+                    using var ms = new System.IO.MemoryStream();
+                    var jsonWriterOptions = new System.Text.Json.JsonWriterOptions
+                    {
+                        Indented = true,
+                        Encoder = System.Text.Encodings.Web.JavaScriptEncoder.UnsafeRelaxedJsonEscaping
+                    };
+                    using (var writer = new System.Text.Json.Utf8JsonWriter(ms, jsonWriterOptions))
+                    {
+                        writer.WriteStartObject();
+                        writer.WritePropertyName("entry");
+                        writer.WriteStartObject();
+                        if (root.TryGetProperty("collectionId", out var col)) writer.WriteString("collectionId", col.GetString());
+                        if (root.TryGetProperty("contents", out var contents)) writer.WriteString("contents", contents.GetString());
+                        if (root.TryGetProperty("transactionId", out var tx)) writer.WriteString("transactionId", tx.GetString());
+                        writer.WriteEndObject();
+                        writer.WriteString("state", "Ready");
+                        writer.WriteEndObject();
+                    }
+                    ms.Position = 0;
+                    // Wrap in a synthetic Response that mimics the original status/headers but with new content.
+                    return new SyntheticResponse(currentResponse, ms.ToArray());
```

Comment:
The Azure `Response` ContentStream is settable.  I'm wondering if we can just set the `currentResponse.contentStream` with your memorystream to avoid wrapping and creating the `SyntheticResponse`. 

https://github.com/Azure/azure-sdk-for-net/blob/88c2ddd06c02a5e88f33c41191b0b6a9ebcce685/sdk/core/Azure.Core/src/Response.cs#L32